app/controllers/mixins/ems_common_angular.rb

```diff
@@ -713,8 +713,10 @@ def build_credentials(ems, mode)
     end
 
     def retrieve_event_stream_selection
-      return "amqp" if @ems.connection_configurations.ceilometer.try(:endpoint).nil? && @ems.connection_configurations.amqp.try(:endpoint)
-      "ceilometer"
+      amqp_auth = @ems.authentications.detect { |a| a.authtype == 'amqp' }
+      return 'amqp' if @ems.connection_configurations.ceilometer.try(:endpoint).nil? && @ems.connection_configurations.amqp.try(:endpoint) && !amqp_auth.nil?
```

Comment:
```ruby
return 'amqp' if @ems.connection_configurations.amqp&.endpoint&.hostname&.present?
```

Reply:
I don't think we need to check both auth and endpoint, it should suffice to have a valid hostname where to connect to, please see example above.